File: javaSpringShip/src/main/java/pl/battleships/javaspringship/service/GameService.java
package pl.battleships.javaspringship.service;

import pl.battleships.api.dto.GameDto;
import pl.battleships.api.dto.GameStatusDto;
import pl.battleships.api.dto.PositionDto;
import pl.battleships.api.dto.ShotStatusDto;

import java.util.List;

public interface GameService {

    void joinTheGame(GameDto game);

    ShotStatusDto opponentShot(String gameId, PositionDto position);

    List<PositionDto> getAllShots(String id);

    GameStatusDto getGameStatus(String gameId);
}

File: kotlinSpringShip/src/main/kotlin/pl/battleships/kotlinspringship/service/GameService.kt
package pl.battleships.kotlinspringship.service

import pl.battleships.api.dto.GameDto
import pl.battleships.api.dto.GameStatusDto
import pl.battleships.api.dto.PositionDto
import pl.battleships.api.dto.ShotStatusDto

interface GameService {

    fun joinTheGame(game: GameDto)

    fun opponentShot(gameId: String, position: PositionDto): ShotStatusDto

    fun getAllShots(id: String): List<PositionDto>

    fun getGameStatus(gameId: String): GameStatusDto
}